keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1224 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**options** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+
+## Summary
+
+
+This KEP builds upon the previously drafted (KEP PR)[https://github.com/kubernetes/enhancements/pull/1592], incorporating additional details and refinements
+based on the feedback received from the reviewers regarding the initial proposal. 
+
+Currently resource allocation on PodSpec is container-centric, allowing users to specify resource requests and limits for each container. The scheduler uses 
+the aggregate of the requested resources by all containers in a pod to find a suitable node for scheduling the pod. The kubelet then enforces these resource
+constraints by translating the requests and limits into corresponding cgroup settings at both the pod and container levels. The existing Pod API doesn’t enable 
+defining resource constraints at pod level, limiting the flexibility and ease of resource management for pods as a whole. 
+
+This KEP proposes extending the Pod API to support Pod-level resource limits and requests for non-extended resources (CPU and Memory), in addition to the 
+existing container-level settings.
+
+
+## Motivation
+
+
+Kubernetes workloads often consist of multiple containers collaborating within a pod. While this separation enables granular monitoring and management of each
+container, it can be challenging to accurately estimate and allocate resources for individual containers, especially for workloads with unpredictable or 
+bursty resource demands. This often leads to over-allocation of resources to ensure that no container experiences resource starvation, as kubernetes currently
+lacks a mechanism for sharing resources between containers easily. 
+
+Introducing pod-level resource requests and limits offers a more simplified resource management for multi-container pods as it is easier to gauge the collective
+resource usage of all containers in a pod rather than predicting each container’s individual needs. This proposal aims to support pod-level resource requests 
+and limits that would allow Kubernetes to manage the total resource consumption of the pod, freeing users from the burden of meticulously allocating resources
+for each container. This simplified approach is particularly valuable for tightly coupled applications, and it will enable Kubernetes to leverage the underlying
+cgroup resource management mechanisms more effectively for better resource utilization for bursty workloads. 
```

Comment:
cgroups is how linux works - If this covers windows, too, then it should maybe mention that up front.

Reply:
Added a section for this.